hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/googlebigquery/avro/GoogleBigQueryAvroStreamWriterFactory.java

```diff
@@ -0,0 +1,75 @@
+package pl.allegro.tech.hermes.consumers.consumer.sender.googlebigquery.avro;
+
+import com.google.api.gax.batching.FlowControlSettings;
+import com.google.api.gax.core.CredentialsProvider;
+import com.google.api.gax.core.ExecutorProvider;
+import com.google.api.gax.core.FixedExecutorProvider;
+import com.google.api.gax.grpc.ChannelPoolSettings;
+import com.google.api.gax.rpc.TransportChannelProvider;
+import com.google.auth.Credentials;
+import com.google.cloud.bigquery.storage.v1.BigQueryWriteClient;
+import com.google.cloud.bigquery.storage.v1.BigQueryWriteSettings;
+import com.google.cloud.bigquery.storage.v1.SchemaAwareStreamWriter;
+import com.google.cloud.bigquery.storage.v1.ToProtoConverter;
+import java.io.IOException;
+import java.util.concurrent.Executors;
+import org.apache.avro.generic.GenericRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.threeten.bp.Duration;
+import pl.allegro.tech.hermes.consumers.config.GoogleBigQueryAvroStreamWriterProperties;
+import pl.allegro.tech.hermes.consumers.consumer.bigquery.GoogleBigQueryStreamWriterFactory;
+
+public class GoogleBigQueryAvroStreamWriterFactory
+    implements GoogleBigQueryStreamWriterFactory<SchemaAwareStreamWriter<GenericRecord>> {
+
+  private static final Logger logger =
+      LoggerFactory.getLogger(GoogleBigQueryAvroStreamWriterFactory.class);
+
+  private final GoogleBigQueryAvroStreamWriterProperties avroStreamWriterProperties;
+  private final Credentials credentials;
+  private final BigQueryWriteClient writeClient;
+  private final ToProtoConverter<GenericRecord> avroToProtoConverter;
+
+  public GoogleBigQueryAvroStreamWriterFactory(
+      GoogleBigQueryAvroStreamWriterProperties avroStreamWriterProperties,
+      CredentialsProvider credentialsProvider,
+      BigQueryWriteSettings writeSettings,
+      ToProtoConverter<GenericRecord> avroToProtoConverter)
+      throws IOException {
+    this.avroStreamWriterProperties = avroStreamWriterProperties;
+    this.credentials = credentialsProvider.getCredentials();
+    this.writeClient = BigQueryWriteClient.create(writeSettings);
+    this.avroToProtoConverter = avroToProtoConverter;
+  }
+
+  public SchemaAwareStreamWriter<GenericRecord> getWriterForStream(String streamName) {
+    try {
+      ExecutorProvider executorProvider =
+          FixedExecutorProvider.create(
+              Executors.newScheduledThreadPool(avroStreamWriterProperties.getPoolSize()));
```

Comment:
Do we need to create a new thread pool for every writer? The default value is 200 threads per stream and number of streams will be equal to the number of gcp partitions for all tables (all topics). The resulting number of threads will be huge